skllm/models/gpt_zero_shot_clf.py

```diff
@@ -19,18 +19,47 @@
 
 
 class _BaseZeroShotGPTClassifier(ABC, BaseEstimator, ClassifierMixin, _OAIMixin):
+    """Base class for zero-shot classifiers.
+
+    Parameters
+    ----------
+    openai_key : Optional[str] , default : None
+        Your OpenAI API key. If None, the key will be read from the SKLLM_CONFIG_OPENAI_KEY environment variable.
+    openai_org : Optional[str] , default : None
+        Your OpenAI organization. If None, the organization will be read from the SKLLM_CONFIG_OPENAI_ORG
+         environment variable.
+    openai_model : str , default : "gpt-3.5-turbo"
+        The OpenAI model to use. See https://beta.openai.com/docs/api-reference/available-models for a list of
+        available models.
+    default_label : Optional[Union[List[str], str]] , default : 'Random'
+        The default label to use if the LLM could not generate a response for a sample. If set to 'Random' a random
+        label will be chosen based on probabilities from the training set.
+    random_state : int, default 42
+        A seed to set for pseudo-random functions, primarily random selection.
+    """
+
     def __init__(
         self,
         openai_key: Optional[str] = None,
         openai_org: Optional[str] = None,
         openai_model: str = "gpt-3.5-turbo",
+        default_label: Optional[Union[List[str], str]] = 'Random',
+        random_state: int = 42,
     ):
         self._set_keys(openai_key, openai_org)
         self.openai_model = openai_model
+        self.default_label = default_label
+        random.seed(random_state)
```

Comment:
I would not set a seed like this as there are plenty of edge cases where this could break.

Imagine an extreme example:
```python
clf1 = Classifier(seed = 41) 
clf2 = Classifier(seed = 42)
clf1.fit(X, y).predict(X) # a wrong seed will be used
```
Even per sklearn official guidelines, preferably, there should be no logic inside the `__init__` method and its only purpose should be to store the arguments.

Reply:
changed